src/index/base.cpp

```diff
@@ -123,8 +123,11 @@ BaseIndex::BaseIndex(std::unique_ptr<interfaces::Chain> chain, std::string name)
 
 BaseIndex::~BaseIndex()
 {
-    Interrupt();
-    Stop();
+    //! Assert Stop() was called before this base destructor. Notification
+    //! handlers call pure virtual methods like GetName(), so if they are still
+    //! being called at this point, they would segfault.
+    LOCK(m_mutex);
+    assert(!m_handler);
```

Comment:
In cf491ea0:

Probably, we should fix the tests that could be depending on this here too. For instance, the `blockfilter_index_init_destroy` unit test isn't interrupting nor stopping the index. 
Then, `coinstatsindex_unclean_shutdown` and `txindex_initial_sync` only call `Stop()` without calling `Interrupt()` first.